src/PeerPoolMgr.cc

```diff
@@ -268,15 +269,46 @@ class PeerPoolMgrsRr: public RegisteredRunner
 {
 public:
     /* RegisteredRunner API */
-    virtual void useConfig() { syncConfig(); }
-    virtual void syncConfig();
+    virtual void useConfig();
+    virtual void syncConfig() { configure(); }
+    virtual void useFullyIndexedStore();
+
+private:
+    void configure();
+    /// whether the Store index was being built in foreground
+    /// at the time of useConfig() call
+    bool waitingForStoreIndex = false;
```

Comment:
We should document the corresponding _state_ (rather than focusing on the conditions that _initiate_ that state):

```suggestion
    /// whether configure() is waiting for the Store index to be built
    bool waitingForStoreIndex = false;
```

... and fix syncConfig() accordingly.